src/ParseQuery.js

```diff
@@ -910,6 +910,38 @@ class ParseQuery {
     });
   }
 
+  /**
+   * Iterates over each result of a query, calling a callback for each one. If
+   * the callback returns a promise, the iteration will not continue until
+   * that promise has been fulfilled. If the callback returns a rejected
+   * promise, then iteration will stop with that error. The items are
+   * processed in an unspecified order. The query may not have any sort order,
+   * and may not use limit or skip.
+   * @param {Function} callback Callback <ul>
+   *   <li>currentObject: The current Parse.Object being processed in the array.</li>
+   *   <li>index: The index of the current Parse.Object being processed in the array.</li>
+   *   <li>query: The query map was called upon.</li>
+   * </ul>
+   *
+   * @param {Object} options Valid options are:<ul>
+   *   <li>useMasterKey: In Cloud Code and Node only, causes the Master Key to
+   *     be used for this request.
+   *   <li>sessionToken: A valid session token, used for making a request on
+   *       behalf of a specific user.
+   * </ul>
+   * @return {Promise} A promise that will be fulfilled once the
+   *     iteration has completed.
+   */
+  async map(callback: (currentObject: ParseObject, index: number, query: ParseQuery) => any, options?: BatchOptions): Promise<Array<any>> {
+    const array = [];
+    let index = 0;
+    await this.each((object) => {
```

Comment:
@acinader This uses `.each()` i'm going to replace it with `find` since `each` has restrictions like no sort/limit/skip etc.

Reply:
well, look at the implementation of each.  I think that each, map, reduce and filter all need to be able to be run against a whole collection.  so at least for a first pass, using the same mechanism as each (or actually using each as you have done) is the way to go.

what you've done just make available in the API what I have been doing a lot in my parse code, so I think what you have is a big improvement by moving the pattern into the SDK.

Reply:
Sweet, i'll add reduce and filter to this PR. That reference link really helped.